src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionHandler.java

```diff
@@ -0,0 +1,158 @@
+/*
+    Copyright 2017 Ericsson AB.
+    For a full list of individual contributors, please see the commit history.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+package com.ericsson.ei.subscriptionhandler;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.ericsson.ei.jmespath.JmesPathInterface;
+import com.ericsson.ei.mongodbhandler.MongoDBHandler;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+/**
+ * This class is responsible to take a aggregatedObject and match it with all
+ * the Subscription Object, to check if any of them is eligible for
+ * notification.
+ * 
+ * @author xjibbal
+ *
+ */
+
+@Component
+public class SubscriptionHandler {
+
+    @Value("${subscriptionCollectionName}")
+    private String SubscriptionCollectionName;
+    @Value("${subscriptionDataBaseName}")
+    private String SubscriptionDataBaseName;
+
+    @Autowired
+    InformSubscription informSubscription;
+
+    @Autowired
+    private MongoDBHandler handler;
+
+    @Autowired
+    private RunSubscription runSubscription;
+
+    @Autowired
+    private JmesPathInterface jmespath;
+
+    private JsonNode requirement;
```

Comment:
What is this?

You cannot have private variables that can be set by some method and used by other method. You need to pass it as a result from methods.